consumer-test/consumerTestController.groovy

```diff
@@ -0,0 +1,144 @@
+        exitPrematurely(1, 'Environment variables CX_INFRA_IT_CF_USERNAME and CX_INFRA_IT_CF_PASSWORD need to be set.')
+    }
+
+    testCaseThreads = [new TestRunnerThread(filePath)]
+
+} else {
+    ConsumerTestUtils.runningLocally = false
+
+    if (changeDoesNotNeedConsumerTesting()) {
+        notifyGithub("success", "No consumer tests necessary.")
+        exitPrematurely(0, 'No consumer tests necessary.')
+    }
+
+    /*
+    In case the build is performed for a pull request TRAVIS_COMMIT is a merge
+    commit between the base branch and the PR branch HEAD. That commit is actually built.
+    But for notifying about a build status we need the commit which is currently
+    the HEAD of the PR branch.
+
+    In case the build is performed for a simple branch (not associated with a PR)
+    In this case there is no merge commit between any base branch and HEAD of a PR branch.
+    The commit which we need for notifying about a build status is in this case simply
+    TRAVIS_COMMIT itself.
+    */
+    ConsumerTestUtils.commitHash = System.getenv('TRAVIS_PULL_REQUEST_SHA') ?: System.getenv('TRAVIS_COMMIT')
+
+    notifyGithub("pending", "Consumer tests are in progress.")
+
+    testCaseThreads = listTestCaseThreads()
+}
+
+testCaseThreads.each { it ->
+    it.start()
+}
+
+//This method will print to console while the test cases are running
+//Otherwise the job will be canceled after 10 minutes without output.
+waitForTestCases(testCaseThreads)
+
+notifyGithub("success", "All consumer tests succeeded.")
+
+
+def listTestCaseThreads() {
+    //Each dir that includes a yml file is a test case
+    def testCases = listYamlInDirRecursive(TEST_CASES_DIR)
+    def threads = []
+    testCases.each { file ->
+        threads << new TestRunnerThread(file.toString())
+    }
+    return threads
+}
+
+def waitForTestCases(threadList) {
+    threadList.metaClass.anyThreadStillAlive = {
+        for (thread in delegate) {
+            if (thread.isAlive()) {
+                return true
+            }
+        }
+        return false
+    }
+
+    def auxiliaryThread = Thread.start {
+        while (threadList.anyThreadStillAlive()) {
+            printOutputOfThreadsIfOneFailed(threadList)
+
+            sleep(AUXILIARY_SLEEP_MS)
+            if (PRINT_LOGS_AFTER_45_MINUTES_COUNTDOWN-- == 0) {
+                threadList.each { thread ->
+                    thread.printOutput()
+                }
+            }
+            if (ConsumerTestUtils.runningLocally) {
+                threadList[0].printRunningStdOut()
+                threadList[0].abortIfSevereErrorOccurred()
+            } else {
+                println "[INFO] Consumer tests are still running."
+            }
+        }
+    }
+    auxiliaryThread.join()
+}
+
+static def printOutputOfThreadsIfOneFailed(threadList) {
+    def failedThread = threadList.find { thread ->
+        thread.exitCode > 0
+    }
+    if (failedThread) {
+        threadList.each { thread ->
```

Comment:
What about `find` rather than each and refusing `n-1` members of the list?

Reply:
This loop should print the output of all other threads (that have not failed).